File: src/main/java/com/nicholasnassar/sfsapi/data/grades/Grade.java
package com.nicholasnassar.sfsapi.data.grades;

import com.nicholasnassar.sfsapi.data.links.GradeLink;
import io.vertx.core.json.JsonObject;

public class Grade {
    private final GradeLink link;

    private final String term, clazz, instructor, score;

    public Grade(GradeLink link, String term, String clazz, String instructor, String score) {
        this.link = link;

        this.term = term;

        this.clazz = clazz;

        this.instructor = instructor;

        this.score = score;
    }

    public String getTerm() {
        return term;
    }

    public String getClazz() {
        return clazz;
    }

    public String getScore() {
        return score;
    }

    public JsonObject asJson() {
        return new JsonObject().put("link", link.asJson()).put("term", term).put("class", clazz)
                .put("instructor", instructor).put("score", score);
    }
}

File: src/main/java/com/nicholasnassar/sfsapi/data/grades/Grade.kt
package com.nicholasnassar.sfsapi.data.grades

import com.nicholasnassar.sfsapi.data.links.GradeLink
import io.vertx.core.json.JsonObject

class Grade(val link: GradeLink, val term: String, val clazz: String, val instructor: String?, val score: String) {
    fun asJson(): JsonObject {
        return JsonObject().put("link", link.asJson()).put("term", term).put("class", clazz)
                .put("instructor", instructor).put("score", score)
    }
}